### PR TITLE
Divya

### LINKED ISSUE #24: emergency_stop _handling
Function to handle the emergency stop

### PR DESCRIPTION
**Pull request by divyalakshmi-padmanaban**
Fixed test functions and added release 2 emergency_stop_handling function

### CODE REVIEW

**Review by ritika2 — COMMENTED** [review]
src/emergency_stop_handling.c doesn't have param[in] and return details.
Also, the local variables used doesn't have proper names. Those are not self-explanatory. Could you either name then properly or give a brief description about them in a single line comment.
Could you please fix these minor problems.

File: src/emergency_stop_handling.c [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		        if((ele[i].moving==1)&&(ele[i].between_floor==1))
+		        {
+			        if(ele[i].direction_up==true)
+			        {
+				        for(j=0;j<ele[i].max_passenger;j++)
+				        {
+				            ele[i].passsenger_arr[j].dest_floor=ele[i].cur_floor+1;
+				            ele[i].moving=k;
+				            ele[i].between_floor=k;
+				            remove_passenger_floor(ele[i].passenger_arr[j]);
```
Comment (MaazJamal) [inline src/emergency_stop_handling.c]: it should call passenger_drop not remove_passenger_floor

File: src/emergency_stop_handling.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+		timer++;
+	for(i=0;i<elevator_count;i++)
+	{
+		if(ele[i].passenegr_count!=0)
+		{
+			status=true;
+			break;
+		}
+		if(ele[i].passenger_count=0){
```
Comment (MaazJamal) [inline src/emergency_stop_handling.c]: should be `ele[i].passenger_count==0`

File: src/emergency_stop_handling.c [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+/**
+ * @file emergency_stop_handling.c
+ *
+ * @brief On account of emergency stop condition,All the elevators re moved to nearest floor.
+ * Passengers are removed from the elevators.
+ *
+ * @author Divyalakshmi Padmanaban  - dpadm049@uottawa.ca	
+ *
+ */
+void emergency_stop_handling(struct elevator ele[],int timer)
```
Comment (MaazJamal) [inline src/emergency_stop_handling.c]: `void emergency_stop_handling(struct elevator ele[],int *timer)` To allow for passing by reference

**Review by MaazJamal — CHANGES_REQUESTED** [review]
There are a few things that need to be changed. PLus there is a bug in the code

File: src/emergency_stop_handling.c (hunk 1 above)
Comment (MaazJamal) [reply src/emergency_stop_handling.c]: the new parameters for passenger_drop are
`int passengers_drop(struct elevator elevator_arr[], int index, int cur_floor, unsigned int event_time);
`
elevator_arr is elevator array 
int index is the current index of the array you are using
int cur floor is the current floor of the elevator. The floor where you are dropping passengers.
unsigned int event time is the timer time the event has happened at.

**Review by MaazJamal — APPROVED** [review]
Okay looks good now.

**Review by Hariprasadm1408 — APPROVED** [review]
Looks good now

**Review by ritika2 — COMMENTED** [review]
Looks good now

**Review by ritika2 — APPROVED** [review]
Looks good now

**Pull request merged**